lib/tasks/full_import.rake

```diff
@@ -1,33 +1,92 @@
-Import = Struct.new("Import", :import_class, :import_method, :folder)
-
-namespace :core do
-  desc "Import all data XMLs from legacy CORE"
-  task :full_import, %i[archive_path] => :environment do |_task, args|
-    archive_path = args[:archive_path]
-    raise "Usage: rake core:full_import['path/to/archive']" if archive_path.blank?
+namespace :import do
+  desc "Run a full import for the institutions listed in the named file on s3"
+  task :full, %i[institutions_csv_name] => :environment do |_task, args|
+    institutions_csv_name = args[:institutions_csv_name]
+    raise "Usage: rake core:full_import['institutions_csv_name']" if institutions_csv_name.blank?
 
     s3_service = Storage::S3Service.new(Configuration::PaasConfigurationService.new, ENV["IMPORT_PAAS_INSTANCE"])
-    archive_io = s3_service.get_file_io(archive_path)
-    archive_service = Storage::ArchiveService.new(archive_io)
+    csv = CSV.parse(s3_service.get_file_io(institutions_csv_name), headers: true)
+    org_count = csv.length
 
-    import_list = [
+    initial_import_list = [
       Import.new(Imports::OrganisationImportService, :create_organisations, "institution"),
       Import.new(Imports::SchemeImportService, :create_schemes, "mgmtgroups"),
       Import.new(Imports::SchemeLocationImportService, :create_scheme_locations, "schemes"),
       Import.new(Imports::UserImportService, :create_users, "user"),
       Import.new(Imports::DataProtectionConfirmationImportService, :create_data_protection_confirmations, "dataprotect"),
       Import.new(Imports::OrganisationRentPeriodImportService, :create_organisation_rent_periods, "rent-period"),
+    ]
+
+    Rails.logger.info("Beginning initial imports for #{org_count} organisations")
+
+    csv.each do |row|
+      archive_path = row[1]
+      archive_io = s3_service.get_file_io(archive_path)
+      archive_service = Storage::ArchiveService.new(archive_io)
+
+      Rails.logger.info("Performing initial imports for organisation #{row[0]}")
+
+      initial_import_list.each do |step|
+        if archive_service.folder_present?(step.folder)
+          step.import_class.new(archive_service).send(step.import_method, step.folder)
+        end
+      end
+    end
+
+    logs_import_list = [
       Import.new(Imports::LettingsLogsImportService, :create_logs, "logs"),
-      # Import.new(Imports::SalesLogsImportService, :create_logs, "logs"),
+      Import.new(Imports::SalesLogsImportService, :create_logs, "logs"),
```

Comment:
seems like a significant change, why weren't we doing this before?

Reply:
We were in the console scripts we were actually running - as far as I can tell the rake task has actually not been used for some time.